### PR TITLE
Update the CarouselView Position setting the CurrentItem

### LINKED ISSUE #7924: [Bug] Setting CarouselView.CurrentItem to an item doesn't display that item
### Description

Setting the `CarouselView.CurrentItem` property, via a data binding, doesn't result in that item being displayed. The code below results in the first item (0) in the collection being displayed, rather than the fourth:

```
        public MonkeysViewModel()
        {
            source = new List<Monkey>();
            CreateMonkeyCollection();

            CurrentItem = Monkeys.Skip(3).FirstOrDefault();
            OnPropertyChanged("CurrentItem");
            //Position = 3;
            //OnPropertyChanged("Position");
        }
```

Note that setting the `Position` property does cause the 4th item in the collection to be displayed.

If you look at the property changed handler for `CurrentItem`, it's meant to cause the `Position` property to be updated:

```
		static void CurrentItemPropertyChanged(BindableObject bindable, object oldValue, object newValue)
		{
			var carouselView = (CarouselView)bindable;
			var args = new CurrentItemChangedEventArgs(oldValue, newValue);
			var command = carouselView.CurrentItemChangedCommand;

			if (command != null)
			{
				var commandParameter = carouselView.CurrentItemChangedCommandParameter;
				if (command.CanExecute(commandParameter))
				{
					command.Execute(commandParameter);
				}
			}

			carouselView.SetValueCore(PositionProperty, GetPositionForItem(carouselView, newValue));
			carouselView.CurrentItemChanged?.Invoke(carouselView, args);
			carouselView.OnCurrentItemChanged(args);
		}
```

Therefore, after setting the `CurrentItem` property I'd expect the `Position` property to be updated and the 4th item in the collection to be displayed, rather than the 1st.

### Steps to Reproduce

1. Run the attached sample.
2. Browse to the "Set current item and position" page.

### Expected Behavior

The Squirrel Monkey card is displayed.

### Actual Behavior

The Baboon card is displayed.

### Basic Information

- Version with issue: 4.3-pre3
- Last known good version: N/A
- IDE: VS Mac 8.3.2
- Platform Target Frameworks: 
  - iOS: 13.0
  - Android: API 28 

### Screenshots

Simulator Screen Shot - iPhone X - 2019-10-10 at 13 26 56

### Reproduction Link

[CarouselViewDemos.zip](https://github.com/xamarin/Xamarin.Forms/files/3712578/CarouselViewDemos.zip)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Update the CarouselView Position setting the **CurrentItem** from a ViewModel. This moves the scroll to the correct position and CurrentItem (object) will be synchronized with Position (int).

### Issues Resolved ### 

- fixes #7924

### API Changes ###

 None

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

None

#### Before
7924-before

_NOTE: The CurrentItem is set in the constructor of the ViewModel, and before the changes of the PR it did not change to the correct position._

#### After
7924-after

### Testing Procedure ###
Launch Core Gallery and navigate to the Issue 7924. If the initial Position is 3, the test passes.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/Items/CarouselView.cs [hunk 1]
```diff
@@ -248,6 +248,18 @@ static int GetPositionForItem(CarouselView carouselView, object item)
 			return 0;
 		}
 
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public object GetItemForPosition(int index)
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply Xamarin.Forms.Core/Items/CarouselView.cs]: Updated the changes to avoid expose nothing new.

**Comment by samhouts** [comment]
@jsuarezruiz Please retarget to 4.4.0. Thanks!

**Comment by samhouts** [comment]
@jsuarezruiz Please retarget to 4.4.0. Thanks!

**Comment by samhouts** [comment]
@jsuarezruiz Please fix conflicts :)

**Comment by samhouts** [comment]
@jsuarezruiz Build failed

https://dev.azure.com/xamarin/public/_build/results?buildId=12149&view=logs&j=bb237aed-7a37-51e1-522f-96005d96ec98&t=829cb1fd-20d5-5ee5-ac21-bd23af5ee57a&l=746

File: Xamarin.Forms.Platform.Android/CollectionView/CarouselViewRenderer.cs [hunk 2]
```diff
@@ -183,6 +184,28 @@ protected override void UpdateAdapter()
 		void UpdateInitialPosition()
 		{
 			_initialPosition = Carousel.Position;
+
+			if (Carousel.CurrentItem != null)
+			{
+				int position = 0;
+
+				var items = Carousel.ItemsSource as IList;
```
Comment (EBusch) [inline Xamarin.Forms.Platform.Android/CollectionView/CarouselViewRenderer.cs]: ItemsSource shouldn't be assumed to be an IList. This has caused other problems in the past with ListViews and CollectionViews. The type is defined as an IEnumerable.

**Pull request merged**